File: src/backend/SDM/SDM.Domain/Entities/Policy.cs
using System;

namespace SDM.Domain.Entities
{
    public class Policy
    {
        public Guid Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string PolicyJson { get; set; } = string.Empty;

        public bool IsEnabled { get; set; } = true;

        public string Category { get; set; } = string.Empty;

        public string Severity { get; set; } = string.Empty;

        public string CommandType { get; set; } = string.Empty;

        /// <summary>Comma-separated EnrollmentType names this policy applies to, e.g. "Corporate,BYOD". Empty means all.</summary>
        public string ApplicableEnrollmentTypes { get; set; } = "Corporate,BYOD";

        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    }
}

File: src/frontend/src/types/policy.ts
export interface Policy {
  id: string;
  name: string;
  policyJson: string;
  isEnabled: boolean;
  category: string;
  severity: string;
  createdOn: string;
}

export interface CreatePolicyRequest {
  name: string;
  policyJson: string;
  isEnabled: boolean;
  category: string;
  severity: string;
}

export interface UpdatePolicyRequest {
  name: string;
  policyJson: string;
  category: string;
  severity: string;
}
